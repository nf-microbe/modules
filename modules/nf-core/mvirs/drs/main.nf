process MVIRS_DRS {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta) , path(contigs)
    tuple val(meta2), path(prophages)

    output:
    tuple val(meta), path("${prefix}.mvirs.dr_info.tsv")    , emit: dr_info
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args   ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    """
    mvirs_drs.py \\
        ${contigs} \\
        ${prophages} \\
        ${prefix}.mvirs.dr_info.tsv \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.mvirs.dr_info.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """
}
