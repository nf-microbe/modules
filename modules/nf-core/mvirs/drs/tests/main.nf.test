nextflow_process {

    name "Test Process MVIRS_DRS"
    script "../main.nf"
    process "MVIRS_DRS"

    tag "modules"
    tag "modules_nfcore"
    tag "mvirs"
    tag "mvirs/index"
    tag "mvirs/oprs"
    tag "mvirs/drs"

    test("fasta.gz") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.contigs.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.mvirs.provirus.fasta.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.contigs.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.mvirs.provirus.fasta.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
