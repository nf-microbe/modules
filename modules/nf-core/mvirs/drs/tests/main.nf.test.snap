{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.dr_info.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b294b3face7c97a2fe35511627a314f7"
                ],
                "dr_info": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.dr_info.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b294b3face7c97a2fe35511627a314f7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-24T13:43:00.083405936"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.dr_info.tsv:md5,de46c5844de598ab7bf4ec4fe4054907"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b294b3face7c97a2fe35511627a314f7"
                ],
                "dr_info": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.dr_info.tsv:md5,de46c5844de598ab7bf4ec4fe4054907"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b294b3face7c97a2fe35511627a314f7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-24T14:26:51.900075963"
    }
}